libs/libarchfpga/src/vib_inf.h

```diff
@@ -0,0 +1,104 @@
+#ifndef VIB_INF_H
+#define VIB_INF_H
+
+#include <functional>
+#include <utility>
+#include <vector>
+#include <unordered_map>
+#include <string>
+#include <map>
+#include <unordered_map>
+#include <limits>
+#include <numeric>
+#include <set>
+#include <unordered_set>
+
+#include "vtr_ndmatrix.h"
+#include "vtr_hash.h"
+#include "vtr_bimap.h"
+#include "vtr_string_interning.h"
+
+#include "logic_types.h"
+#include "clock_types.h"
+
+//#include "physical_types.h"
+
+
+
+/* for vib tag */
+struct t_seg_group {
+    std::string name; 
+    int seg_index;
+    int track_num;
+};
+
+enum e_multistage_mux_from_or_to_type {
+    PB = 0,
+    SEGMENT,
+    MUX
+};
+
+struct t_from_or_to_inf {
+    std::string type_name;
+    e_multistage_mux_from_or_to_type from_type;  //from_or_to_type
+    int type_index = -1;
+    int phy_pin_index = -1;
+    char seg_dir = ' ';
+    int seg_index = -1;
+};
+
+struct t_first_stage_mux_inf {
+    std::string mux_name; 
+    std::vector<t_from_or_to_inf> froms;
+};
+
+struct t_second_stage_mux_inf : t_first_stage_mux_inf {
+    std::vector<t_from_or_to_inf> to;    // for io type, port[pin] may map to several sinks
+};
+
+// struct t_vib_inf {
+//     std::string name;           /* vib name */
+//     std::string pbtype_name;    /* pbtype name of vib */
+//     int seg_group_num;          /* seg group number of vib */
+//     int switch_idx;             /* vib switch index */
+//     std::vector<t_seg_group> seg_groups;
+//     std::vector<t_first_stage_mux_inf> first_stages;
+//     std::vector<t_second_stage_mux_inf> second_stages;
+// };
+
```

Comment:
Need a figure in ASICII format to explanation the concepts ``pbtype_name``, ``seg_group_num``, ``switch_idx`` etc.